test/Fixtures.purs

```diff
@@ -321,6 +321,8 @@ proposedProtocolParameterUpdates1 = ProposedProtocolParameterUpdates $
         , maxBlockExUnits: Just
             { mem: BigInt.fromInt 1, steps: BigInt.fromInt 1 }
         , maxValueSize: Just $ UInt.fromInt 1
+        , collateralPercentage: Nothing -- Just $ UInt.fromInt 140
+        , maxCollateralInputs: Nothing -- Just $ UInt.fromInt 10
```

Comment:
@klntsky I can't figure out why the unit test `Deserialization is inverse to serialization #4` fails when I set the two new fields to a `Just` value. The error msg is `Cannot deserialize bytes`. meanwhile, if the fields are `Nothing`, the test passes

Reply:
@pb99u069 use `spy` function to debug this (you can detect where decoding fails by inserting it in a pure expression)

Reply:
If you print the error message CSL returns in `fromBytes`, it seems confused and thinks the value we pass as a collateral_percentage is a key in some map. So either we're doing something wrong on the to_bytes/CSL building side, or CSL has a bug

Reply:
The functions `set_collateral_percentage` and `set_max_collateral_inputs` [here](https://github.com/Emurgo/cardano-serialization-lib/blob/fa2a9c94bcd1e3a7353852915a0f2e666ec0027f/rust/src/lib.rs#L2690) differ from all the other setters, as they don't use `clone()` at the end. Maybe that's our problem?

Reply:
I saw that too but there are other cases that aren't cloned, `max_tx_size` for example

Reply:
@jy14898 
> it seems confused and thinks the value we pass as a collateral_percentage is a key in some map

Could you please post the message for me to find where is it thrown from?

Reply:
Definitely a bug with CSL:

```
> ppu = lib.ProtocolParamUpdate.new()
ProtocolParamUpdate { ptr: 1668248 }
> ppu.set_collateral_percentage(42)
undefined
> ppu.collateral_percentage()
42
> lib.ProtocolParamUpdate.from_bytes(ppu.to_bytes()).collateral_percentage()
undefined
```

Reply:
@klntsky I used the following to always print the error, as we hush it in purescript. But the error is misleading due to nested decoding, the crux of the problem is as I posted above

```diff
--- a/src/Internal/Deserialization/FromBytes.js
+++ b/src/Internal/Deserialization/FromBytes.js
@@ -11,6 +11,7 @@ const fromBytes = name => helper => bytes => {
   try {
     return helper.valid(lib[name].from_bytes(bytes));
   } catch (e) {
+    console.log(name + ".from_bytes() raised " + e);
     return helper.error(name + ".from_bytes() raised " + e);
   }
 };
```

Reply:
Made an issue
https://github.com/Emurgo/cardano-serialization-lib/issues/564

Reply:
https://github.com/Emurgo/cardano-serialization-lib/pull/565

Reply:
Yep, but the release hasn't been cut yet (I hope it will be soon)